aws/resource_aws_s3_bucket_analytics_configuration.go

```diff
@@ -0,0 +1,437 @@
+																Default:      s3.AnalyticsS3ExportFileFormatCsv,
+																ValidateFunc: validation.StringInSlice([]string{s3.AnalyticsS3ExportFileFormatCsv}, false),
+															},
+															"prefix": {
+																Type:     schema.TypeString,
+																Optional: true,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+var filterAtLeastOneOfKeys = []string{"filter.0.prefix", "filter.0.tags"}
+
+func schemaWithAtLeastOneOf(schema *schema.Schema, keys []string) *schema.Schema {
+	schema.AtLeastOneOf = keys
+	return schema
+}
+
+func resourceAwsS3BucketAnalyticsConfigurationPut(d *schema.ResourceData, meta interface{}) error {
+	s3conn := meta.(*AWSClient).s3conn
+
+	bucket := d.Get("bucket").(string)
+	name := d.Get("name").(string)
+
+	log.Printf("[DEBUG] S3 bucket %q, add analytics configuration %q", bucket, name)
+
+	analyticsConfiguration := &s3.AnalyticsConfiguration{
+		Id:                   aws.String(name),
+		Filter:               expandS3AnalyticsFilter(d.Get("filter").([]interface{})),
+		StorageClassAnalysis: expandS3StorageClassAnalysis(d.Get("storage_class_analysis").([]interface{})),
+	}
+
+	input := &s3.PutBucketAnalyticsConfigurationInput{
+		Bucket:                 aws.String(bucket),
+		Id:                     aws.String(name),
+		AnalyticsConfiguration: analyticsConfiguration,
+	}
+
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := s3conn.PutBucketAnalyticsConfiguration(input)
+		if err != nil {
+			if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if isResourceTimeoutError(err) {
+		_, err = s3conn.PutBucketAnalyticsConfiguration(input)
+	}
+	if err != nil {
+		return fmt.Errorf("Error adding S3 analytics configuration: %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", bucket, name))
+
+	return resourceAwsS3BucketAnalyticsConfigurationRead(d, meta)
+}
+
+func resourceAwsS3BucketAnalyticsConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).s3conn
+
+	bucket, name, err := resourceAwsS3BucketAnalyticsConfigurationParseID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.Set("bucket", bucket)
+	d.Set("name", name)
+
+	input := &s3.GetBucketAnalyticsConfigurationInput{
+		Bucket: aws.String(bucket),
+		Id:     aws.String(name),
+	}
+
+	log.Printf("[DEBUG] Reading S3 bucket analytics configuration: %s", input)
+	output, err := conn.GetBucketAnalyticsConfiguration(input)
+	if err != nil {
+		if isAWSErr(err, s3.ErrCodeNoSuchBucket, "") || isAWSErr(err, "NoSuchConfiguration", "The specified configuration does not exist.") {
+			log.Printf("[WARN] %s S3 bucket analytics configuration not found, removing from state.", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
```

Comment:
Nit: Its generally a good idea to return error context about the error being returned for easier troubleshooting by operators and maintainers:

```suggestion
		return fmt.Errorf("error getting S3 Bucket Analytics Configuration (%s): %w", d.Id(), err)
```